File: cpp/design_patterns/constructor/src/BreaksDummy.cpp
#include <stdio.h>
#include "BreaksDummy.h"

BreaksDummy::BreaksDummy() {
}

BreaksDummy::~BreaksDummy() {
}

void BreaksDummy::Apply() {
    printf("%s: Applying breaks\n", __func__);
}

void BreaksDummy::Release() {
    printf("%s: Release breaks\n", __func__);
}

Brakes::~Brakes() {
    printf("%s\n", __func__);
}

File: java/design/contructor/src/contructor/BreaksDummy.java
package contructor;

public class BreaksDummy implements Breaks {

  private Logger logger;

  public static BreaksDummy GetInstance(Logger logger) {
    return new BreaksDummy(logger);
  }
  
  private BreaksDummy(Logger logger) {
    this.logger = logger;
  }

  @Override
  public void Apply() {
    logger.Log("BreaksDummy: Apply");
  }

  @Override
  public void Release() {
    logger.Log("BreaksDummy: Release");
  }

}
